modules/charon-core/src/main/java/org/wso2/charon3/core/schema/SCIMResourceTypeSchema.java

```diff
@@ -39,6 +45,9 @@ private SCIMResourceTypeSchema(List<String> schemas, AttributeSchema[] attribute
                 this.attributeList.add(attributeSchema);
             }
         }
+        // assuming that a "ResourceTypeSchema" in most cases only has ONE schema uri,
+        // otherwise it would be considered a resource (which are subclasses of AbstractSCIMObject)
+        existingResourceSchemas.put(schemas.get(0), this);
```

Comment:
'existingResourceSchemas' map should be thread-safe here since SCIMResourceTypeSchema constructor can be called through createSCIMResourceSchema method by multiple threads.